src/spec-node/devContainersSpecCLI.ts

```diff
@@ -200,6 +201,12 @@ async function provision({
 	const addCacheFroms = addCacheFrom ? (Array.isArray(addCacheFrom) ? addCacheFrom as string[] : [addCacheFrom]) : [];
 	const additionalFeatures = additionalFeaturesJson ? jsonc.parse(additionalFeaturesJson) as Record<string, string | boolean | Record<string, string | boolean>> : {};
 	const providedIdLabels = idLabel ? Array.isArray(idLabel) ? idLabel as string[] : [idLabel] : undefined;
+
+	const cwd = workspaceFolder || process.cwd();
+	const cliHost = await getCLIHost(cwd, loadNativeModule);
+	const secretsP = readSecretsFromFile({ secretsFile, cliHost });
+	const log = async (text: string) => process.stderr.write(await maskSecrets(secretsP, text));
```

Comment:
The output here can be JSON. That creates the edge case of a secret matching a JSON property (or some other part of the JSON syntax). Maybe add a helper function to log.ts: `function replaceAllLog(origin: LogHandler, values: string[], replacement: string): LogHandler`. (See `createCombinedLog` in that file for an example.)

`replaceAllLog` can apply the replacements to the log event's text and pass a modified log event on. You can then wrap the `handler` in `createLogFrom` in devContainer.ts with `replaceAllLog`.

Reply:
Maybe (untested):
```ts
export function replaceAllLog(origin: LogHandler, values: string[], replacement: string): LogHandler {
	const r = new RegExp(values.map(v => v.replace(/[.*+?^${}()|[\]\\]/g, '\\$&')).join('|'), 'g');
	return {
		event: e => {
			if ('text' in e) {
				origin.event({
					...e,
					text: e.text.replace(r, replacement),
				});
			} else {
				origin.event(e);
			}
		}
	};
}
```

`stepDetails` might need the same treatment.

Reply:
I just committed a version of my attempt.. (before your last comment). Would you mind taking a look?

Reply:
Implemented the suggestion. Thank you